contentprovider.go

```diff
@@ -234,6 +241,24 @@ func (p *contentProvider) fillContentMatches(ms []*candidateMatch) []LineMatch {
 	return result
 }
 
+func getLines(data []byte, newLines []uint32, low, high int) []byte {
+	if high < 0 || low >= len(newLines) || len(newLines) == 0 {
+		return nil
+	}
+
+	var startIndex uint32
+	if low < 0 {
+		startIndex = 0
+	} else {
+		startIndex = newLines[low] + 1
+	}
+
+	if high >= len(newLines) {
+		return data[startIndex:]
+	}
+	return data[startIndex:newLines[high]]
```

Comment:
nice, this part LGTM. Now before this we just need to make sure we are observing the correct invariants.